cmd/influxd/run.go

```diff
@@ -128,35 +106,92 @@ func parseConfig(path, hostname string) *Config {
 	return config
 }
 
-// creates and initializes a broker at a given path.
-func openBroker(path, addr string) *messaging.Broker {
+// creates and initializes a broker.
+func openBroker(path string, u *url.URL, initializing bool, joinURLs []*url.URL) *messaging.Broker {
+	// Ignore if there's no existing broker and we're not initializing or joining.
+	if !fileExists(path) && !initializing && len(joinURLs) == 0 {
```

Comment:
In other words, I think you could just have replaced the entire check of `if b != nil` with this logic, if it was outside the function. It's not a big deal though, since the outcome is the same.